NorthstarDLL/httprequesthandler.cpp

```diff
@@ -0,0 +1,504 @@
+{
+	if (IsRunning())
+	{
+		spdlog::warn("HttpRequestHandler::StartHttpRequestHandler was called while IsRunning() is true");
+		return;
+	}
+
+	m_bIsHttpRequestHandlerRunning = true;
+	spdlog::info("HttpRequestHandler started.");
+}
+
+void HttpRequestHandler::StopHttpRequestHandler()
+{
+	if (!IsRunning())
+	{
+		spdlog::warn("HttpRequestHandler::StopHttpRequestHandler was called while IsRunning() is false");
+		return;
+	}
+
+
+	m_bIsHttpRequestHandlerRunning = false;
+	spdlog::info("HttpRequestHandler stopped.");
+}
+
+bool IsHttpDestinationHostAllowed(const std::string& host, std::string& outHostname, std::string& outAddress, std::string& outPort)
+{
+	bool bAllowed = false;
+
+	CURLU* url = curl_url();
+	if (!url)
+	{
+		spdlog::error("Failed to call curl_url() for http request.");
+		return false;
+	}
+
+	if (curl_url_set(url, CURLUPART_URL, host.c_str(), CURLU_DEFAULT_SCHEME) != CURLUE_OK)
+	{
+		spdlog::error("Failed to parse destination URL for http request.");
+
+		curl_url_cleanup(url);
+		return false;
+	}
+
+	char* urlHostname = nullptr;
+	if (curl_url_get(url, CURLUPART_HOST, &urlHostname, 0) != CURLUE_OK)
+	{
+		spdlog::error("Failed to parse hostname from destination URL for http request.");
+
+		curl_url_cleanup(url);
+		return false;
+	}
+
+	char* urlScheme = nullptr;
+	if (curl_url_get(url, CURLUPART_SCHEME, &urlScheme, CURLU_DEFAULT_SCHEME) != CURLUE_OK)
+	{
+		spdlog::error("Failed to parse scheme from destination URL for http request.");
+
+		curl_url_cleanup(url);
+		curl_free(urlHostname);
+		return false;
+	}
+
+	char* urlPort = nullptr;
+	if (curl_url_get(url, CURLUPART_PORT, &urlPort, CURLU_DEFAULT_PORT) != CURLUE_OK)
+	{
+		spdlog::error("Failed to parse port from destination URL for http request.");
+
+		curl_url_cleanup(url);
+		curl_free(urlHostname);
+		curl_free(urlScheme);
+		return false;
+	}
+
+	// Resolve the hostname into an address.
+	addrinfo* result;
+	addrinfo hints;
+	std::memset(&hints, 0, sizeof(addrinfo));
+	hints.ai_family = AF_UNSPEC;
+
+	if (getaddrinfo(urlHostname, urlScheme, &hints, &result) != 0)
+	{
+		spdlog::error("Failed to resolve http request destination {} using getaddrinfo().", urlHostname);
+
+		curl_url_cleanup(url);
+		curl_free(urlHostname);
+		curl_free(urlScheme);
+		curl_free(urlPort);
+		return false;
+	}
+
+	bool bFoundIPv6 = false;
+	sockaddr_in* sockaddr_ipv4 = nullptr;
+	for (addrinfo* info = result; info; info = info->ai_next)
+	{
+		if (info->ai_family == AF_INET)
+		{
+			sockaddr_ipv4 = (sockaddr_in*)info->ai_addr;
+			break;
+		}
+
+		bFoundIPv6 = bFoundIPv6 || info->ai_family == AF_INET6;
+	}
+
+	if (sockaddr_ipv4 == nullptr)
+	{
+		if (bFoundIPv6)
+		{
+			spdlog::error("Only IPv4 destinations are supported for HTTP requests. To allow IPv6, launch the game using -allowlocalhttp.");
+		}
+		else
+		{
+			spdlog::error("Failed to resolve http request destination {} into a valid IPv4 address.", urlHostname);
+		}
+		
+		goto CLEANUP;
```

Comment:
Bad use of goto, use inlines instead. Using inlines will make the code more readable, and likely generate more performant code as well.